### ISSUE TITLE
Debugger pauses on the wrong line in Windows

### ISSUE THREAD

**Issue by 4amVim** [Dart-Code/Dart-Code#3337]
**Describe the bug**
Debugger pauses on the wrong line

**To Reproduce**
1. Write some code with a call to `debugger();`
2. Run it, and wait for the pause

**Expected behavior**
The line with `debugger(when:true);` is the line in which it should pause.

**Screenshots**
In the third last line is the `debugger(when:true);` statement which triggers the pause, but the pause actually occurs later:
![image](https://user-images.githubusercontent.com/4429609/117967784-187da400-b343-11eb-9a25-41f82755541b.png)
Moreover, its not just a line issue, because the variable `e` is out of scope by that point:
![image](https://user-images.githubusercontent.com/4429609/117968433-cee18900-b343-11eb-8e64-c3205a47ec0b.png)

**Versions (please complete the following information):**
 - VS Code version: 1.56.1
 - Dart extension version: 3.22.0
 - Dart/Flutter SDK version: 2.12.3/2.0.6
- OS: Windows 10

Note that for some reason the issue doesn't occur on macOS

[Logs](https://github.com/Dart-Code/Dart-Code/files/6465899/Dart-Code-Log-2021-04-03.17-06-36.txt)

**Comment by DanTup** [comment]
Are you able to share a simple example that you can repro this with? I tried to reproduce, but it seemed to work ok:

<img width="404" alt="Screenshot 2021-05-12 at 15 23 58" src="https://user-images.githubusercontent.com/1078012/117991633-25e06180-b336-11eb-85d6-fbade1d37676.png">

I checked the log, but it didn't include the script source so it's not easy to see what's going on. Although I noticed on line 122 you have a closing `}` that's intended the same as the one above - is that the close of the `initialize` method, or is it possible you've modified this code since running (that could upset things as the source wouldn't match).

**Comment by 4amVim** [comment]
It is indeed the closeof `initialize`, and no I didn't modify the code after running.

Anyhow, my apologies for not using a simple example to repro, as that would also make the log much more palattable, I shall do that soon(~12 hours).

**Comment by 4amVim** [comment]
Okay, caught it

Here's what the my `main.dart`, and of note is only the function `_tester()`:
```dart
import 'dart:developer';

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: MyHomePage());
}

class MyHomePage extends StatefulWidget {
  MyHomePage();
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    _tester();
    setState(() => _counter++);
  }

  void _tester() {
    try {
      throw 'a';
    } catch (e) {
      print("if you see me, pause should've triggered");
      debugger(when: true);
      // print('if you uncomment me, the this is where we get paused');
    }
    print('but the debugger pauses here !?!');
  }

  @override
  Widget build(BuildContext context) => Scaffold(
        appBar: AppBar(title: Text('Hi')),
        body: Center(
            child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: <Widget>[
              Text('You have pushed the button this many times:'),
              Text('$_counter', style: Theme.of(context).textTheme.headline4),
            ])),
        floatingActionButton: FloatingActionButton(
          onPressed: _incrementCounter,
          tooltip: 'Increment',
          child: Icon(Icons.add),
        ), // This trailing comma makes auto-formatting nicer for build methods.
      );
}
```
[full-project here](https://github.com/4amVim/flutterStarter/tree/no-traffic-rulez-for-debugger)

Interestingly, the bug pops up for android but not web, and it goes like this:

https://user-images.githubusercontent.com/4429609/118058182-3ed72a00-b3ab-11eb-9947-6aa1d1f74dba.mp4
[commented.txt](https://github.com/Dart-Code/Dart-Code/files/6469585/commented.txt)


https://user-images.githubusercontent.com/4429609/118058325-983f5900-b3ab-11eb-82e0-2d40d511c1fc.mp4
[uncommented.txt](https://github.com/Dart-Code/Dart-Code/files/6469591/uncommented.txt)

(Instead of capturing pics I thought video would pair better with the respective logs. They all are also there in the linked repository)

**Comment by DanTup** [comment]
Thanks! I was able to repro with your project, and actually able to strip it down to a much simpler case. I think my mistake previously was using the Chrome device to repro (this uses a different debugger, and I assumed it was more likely to have a bug like this than the VM, as I'd found [a similar bug](https://github.com/dart-lang/sdk/issues/45544) recently).

I've filed an SDK issue at https://github.com/dart-lang/sdk/issues/46006 with some additional notes on this, as it'll need fixing there. Thanks!
